First version of diff command for roles and users

ArChrisVa opened the pull request:
Minor changes to role comparison
New command accepts string arguments as the roles and
not role ID's  anymore. Color seems kinda weird.

New sub-command compare that compares member's
date of arrival, date since last nitro boost,
names, status and roles.

Example: diff member Query#1234 random#4567

Also added thumbnail and color in embed builders

Commit:
Minor changes to role comparison

New command accepts as arguments to roles as strings and
not role ID's .

New sub-command compare that compares member's
date of arrival, date since last nitro boost,
names, status and roles.

Example: diff member Query#1234 random#4567

Also added thumbnail and color to the builders

Chew commented on src/main/java/pw/chew/chewbotcca/commands/DiffCommand.java:
```diff
@@ -62,44 +66,68 @@ protected void execute(CommandEvent event) {
             }
 
-            Role base = event.getGuild().getRoleById(args[0]);
+            String baseString = args[0].toLowerCase();
```
The reason these were IDs is because role names can have spaces. If someone wanted to compare roles "joe bob" and "bob joe", this logic would only compare joe to bob.

ArChrisVa replied:
Thats something new to me. I didn't know about spaces between names and roles. I am thinking of using a seperator like "/ " instead of space between names or roles. If roles have the same name , this wouldn't work ... But how many discord users have developer mode enabled or even know about this, so that they can copy id of a role? Thats a problem.

Chew commented on src/main/java/pw/chew/chewbotcca/commands/DiffCommand.java:
```diff
@@ -62,44 +66,68 @@ protected void execute(CommandEvent event) {
+                        "Base role: " + base.getName()  + "\n" + "Compare role: " + compare.getName())
+                    .setColor(new Color(96, 70, 184));
+            try {
+                embed.setThumbnail("https://c1.wallpaperflare.com/preview/634/485/143/analysis-antique-background-balance.jpg");
```
Don't really see the use of this, sell me on it.

ArChrisVa replied:
Embed bulder color -> removed.

Chew commented on src/main/java/pw/chew/chewbotcca/commands/DiffCommand.java:
```diff
@@ -62,44 +66,68 @@ protected void execute(CommandEvent event) {
+            try {
+                embed.setThumbnail("https://c1.wallpaperflare.com/preview/634/485/143/analysis-antique-background-balance.jpg");
+            }catch (Exception e){
+                Sentry.captureException(e);
```
Unnecessary try/catch.

ArChrisVa replied:
The reason is that wallpaperflare might remove this image in future, hence an error would occur. 
I am gonna remove thumbnails too and keep it simple.

Chew replied:
Even if it's removed, it would just not appear on Discord. This would only raise an exception if the URL formatting was invalid.

ArChrisVa replied:
ok then, its gonna be removed tho, i think a cleaner and simpler embed builder is better.

Chew commented on src/main/java/pw/chew/chewbotcca/commands/DiffCommand.java:
```diff
@@ -62,44 +66,68 @@ protected void execute(CommandEvent event) {
+            // compare names
             if (base.getName().equals(compare.getName())) {
-                embed.addField("Name", "*Nothing to compare. Names are identical*", true);
+                embed.addField("Try again with different roles", "*Nothing to compare. Roles are identical*", true);
```
Can't necessarily do this because 2 different roles can have the same name.

Chew commented on src/main/java/pw/chew/chewbotcca/commands/DiffCommand.java:
```diff
@@ -62,44 +66,68 @@ protected void execute(CommandEvent event) {
-                    args[1] + ": " + compare.getName();
-                embed.addField("Name", name, true);
+                String name = base.getName() + "\n" +
+                        compare.getName();
```
Should be added to the line above

Chew commented on src/main/java/pw/chew/chewbotcca/commands/DiffCommand.java:
```diff
@@ -62,44 +66,68 @@ protected void execute(CommandEvent event) {
-                embed.addField("Color", "*Nothing to compare. Colors are identical*", true);
+            // compare colors ( RGB values )
+            if (base.getColorRaw() == compare.getColorRaw()) {
+                embed.addField("Color 🔴🟠🟡🟢🔵🟣🟤⚫⚪", "*Nothing to compare. Colors are identical*", true);
```
I don't understand the emoji on the field names, sell me on this.

ArChrisVa replied:
Would you like to completely remove color comparison? I think no one would be intrested in rgb values of a role's color.

Chew replied:
Would probably just show HEX, RGB isn't really needed.

Chew commented on src/main/java/pw/chew/chewbotcca/commands/DiffCommand.java:
```diff
@@ -131,11 +160,135 @@ protected void execute(CommandEvent event) {
+            Member member2 = null;
+
+            try {
+                member1 = event.getGuild().getMemberByTag(args[0]);
```
Same logic as before. Names can have spaces. Comparing "joe bob#0001" and "bob joe#0002" won't work, it would end up comparing "joe" and "bob#0001".
I am actually unable to test this because the people on my test server all have spaces. This should be handled by either handing quotes, or just doing mention or ID.

ArChrisVa replied:
Again here i am gonna test a different command format in names section

Chew commented on src/main/java/pw/chew/chewbotcca/commands/DiffCommand.java:
```diff
@@ -131,11 +160,135 @@ protected void execute(CommandEvent event) {
+
+            EmbedBuilder embed = new EmbedBuilder()
+                    .setTitle("Comparing members")
+                    .setDescription("Comparing Time joined,Time boosted , Roles, and other information.\n" +
```
```suggestion
                    .setDescription("Comparing time joined, time boosted, roles, and other information.\n" +
```

ArChrisVa replied:
fixed locally

Chew commented on src/main/java/pw/chew/chewbotcca/commands/DiffCommand.java:
```diff
@@ -131,11 +160,135 @@ protected void execute(CommandEvent event) {
+            EmbedBuilder embed = new EmbedBuilder()
+                    .setTitle("Comparing members")
+                    .setDescription("Comparing Time joined,Time boosted , Roles, and other information.\n" +
+                        "First member: " + member1.getUser().getName() + "\n" + "Second member : " + member2.getUser().getName())
```
```suggestion
                        "Base member: " + member1.getUser().getName() + "\n" + "Compare member: " + member2.getUser().getName())
```

ArChrisVa replied:
fixed locally

Chew commented on src/main/java/pw/chew/chewbotcca/commands/DiffCommand.java:
```diff
@@ -131,11 +160,135 @@ protected void execute(CommandEvent event) {
+                try {
+                    embed.setThumbnail("https://c1.wallpaperflare.com/preview/634/485/143/analysis-antique-background-balance.jpg");
+                }catch (Exception e){
+                    Sentry.captureException(e);
```
See above discussion for this

ArChrisVa replied:
I answered above.

Chew commented on src/main/java/pw/chew/chewbotcca/commands/DiffCommand.java:
```diff
@@ -131,11 +160,135 @@ protected void execute(CommandEvent event) {
+                    boostingTime2 = "Currently not boosting";
+                }
+
+                embed.addField("💎 Nitro Boosting 💎", boostingTime1 + "\n" + boostingTime2, true);
```
```suggestion
                embed.addField("💎 Server Boosting 💎", boostingTime1 + "\n" + boostingTime2, true);
```

ArChrisVa replied:
fixed locally

Chew commented on src/main/java/pw/chew/chewbotcca/commands/DiffCommand.java:
```diff
@@ -18,11 +18,14 @@
 import net.dv8tion.jda.api.entities.Role;
 
-import java.util.ArrayList;
+import java.awt.*;
```
Please avoid star imports

Chew reviewed (changes requested):
Thank you for your contributions! Just have a few questions and suggestions.

Commit:
*diff command update*
1. fixed null when color is default color
2. Color is now visible as hex value
3.  not sure about line 101 and 102 method getRed() warning.
4. added information comparison in roles( similar format to roleinfo command )
5. removed most emojis and colors
6. simpler embedbuilders
7. diff member command takes as input mentioned members.

Commit:
reformat

Commit:
reformat

Chew reviewed (approved):
Going to go ahead and merge, but perform my own changes before putting the final product in the bot. Thank you for your contribution!
